### PR TITLE
support scala3 for google-cloud-storage connector

### PR DESCRIPTION
**Pull request by pjfanning**
small compile problem in one class (scala 3 only)

needed to add back some code accidentally reverted in https://github.com/apache/incubator-pekko-connectors/commit/8674ff591172b62986617d6eb61a4ff43a92d417

### CODE REVIEW

File: google-cloud-storage/src/main/scala/org/apache/pekko/stream/connectors/googlecloud/storage/impl/Formats.scala [hunk 1]
```diff
@@ -110,8 +110,8 @@ object Formats extends DefaultJsonProtocol {
       StorageObjectJson(readOnlyFields, writeableFields)
     }
     override def write(obj: StorageObjectJson): JsValue = {
-      val fields1 = obj.readable.toJson.asJsObject.fields
-      val fields2 = obj.writeable.toJson.asJsObject.fields
+      val fields1 = spray.json.enrichAny(obj.readable).toJson.asJsObject.fields
+      val fields2 = spray.json.enrichAny(obj.writeable).toJson.asJsObject.fields
```
Comment (He-Pin) [inline google-cloud-storage/src/main/scala/org/apache/pekko/stream/connectors/googlecloud/storage/impl/Formats.scala:114]: What's the error here?

Comment (pjfanning) [reply google-cloud-storage/src/main/scala/org/apache/pekko/stream/connectors/googlecloud/storage/impl/Formats.scala:114]: won't compile - it says that `.toJson` is not a method on the relevent classes. The implicit conversion `enrichAny` is not happening - so I have to do it explicitly.

Scala 3 compiler, unfortunately, still has some bugs.

Comment (He-Pin) [reply google-cloud-storage/src/main/scala/org/apache/pekko/stream/connectors/googlecloud/storage/impl/Formats.scala:114]: how about import the implicit conversation in scope explicitly，will that work？

Comment (pjfanning) [reply google-cloud-storage/src/main/scala/org/apache/pekko/stream/connectors/googlecloud/storage/impl/Formats.scala:114]: I tried basically everything. The code I have here is what the compiler generates in Scala 2 anyway.

Comment (pjfanning) [reply google-cloud-storage/src/main/scala/org/apache/pekko/stream/connectors/googlecloud/storage/impl/Formats.scala:114]: I'll try again

Comment (He-Pin) [reply google-cloud-storage/src/main/scala/org/apache/pekko/stream/connectors/googlecloud/storage/impl/Formats.scala:114]: wired, let's me try it too💖💖

Comment (pjfanning) [reply google-cloud-storage/src/main/scala/org/apache/pekko/stream/connectors/googlecloud/storage/impl/Formats.scala]: looks like you were right, sorting out the imports fixed the problem - strange how Scala 2 compiler finds the implicit much more easily than Scala compiler - but it's good to know that the Scala3 compiler doesn't have a bug

**Review by He-Pin — APPROVED** [review]
lgtm

**Pull request merged**